Additional platform support for Amazon Linux

Only the OS specific tasks for CentOS need to be executed on Amazon Linux to have support for Amazon Linux

Hi,

Thanks for adding these changes.
I have tested the changes in branch origin/issue/131, but it fails on this branch:

**How I get it working:**
- I installed your role via ansible galaxy (https://github.com/fubarhouse/ansible-role-golang/archive/2.8.0.tar.gz)
- I added the same changes as described in the possible patch
- it works on Amazon Linux -> needed dependencies are installed via yum and go can be installed

**How it fails:**
- I cloned your repository as a role (with possible other differences compared with ansible galaxy download)
- checkout of this branch (origin/issue/131) with same patch
- run the same playbook as with ansible-galaxy example, but with this local role now fails
I do not understand the error, but I think this error does not have something to do with the proposed patch for installing the needed dependencies?
When I manually remove a dependency, the role will install now the dependency by using yum (which was not working before the patch), in the task "Install dependencies", but the task "Run get commands" fails. I have this issue:

```failed: [testNode] (item={u'url': u'github.com/golang/dep/cmd/dep', u'name': u'dep'}) => {"ansible_loop_var": "item", "changed": false, "cmd": ["/usr/local/go/bin/go", "get", "-u", "github.com/golang/dep/cmd/dep"], "delta": "0:00:06.122752", "end": "2019-07-29 08:40:41.436575", "item": {"name": "dep", "url": "github.com/golang/dep/cmd/dep"}, "msg": "non-zero return code", "rc": 2, "start": "2019-07-29 08:40:35.313823", "stderr": "go: finding github.com/golang/dep/cmd/dep latest\ngo: finding github.com/golang/dep/cmd latest\ngo: finding github.com/golang/dep v0.5.4\ngo: downloading github.com/golang/dep v0.5.4\ngo: extracting github.com/golang/dep v0.5.4\ngo: finding github.com/sdboyer/constext latest\ngo: finding github.com/jmank88/nuts v0.3.0\ngo: finding github.com/boltdb/bolt v1.3.1\ngo: finding github.com/pkg/errors v0.8.1\ngo: finding github.com/Masterminds/semver v1.4.2\ngo: finding github.com/pelletier/go-toml v1.4.0\ngo: finding github.com/armon/go-radix v1.0.0\ngo: finding github.com/nightlyone/lockfile latest\ngo: downloading github.com/sdboyer/constext v0.0.0-20170321163424-836a14457353\ngo: extracting github.com/sdboyer/constext v0.0.0-20170321163424-836a14457353\ngo: downloading github.com/jmank88/nuts v0.3.0\ngo: extracting github.com/jmank88/nuts v0.3.0\ngo: downloading github.com/armon/go-radix v1.0.0\ngo: finding golang.org/x/sync/errgroup latest\ngo: finding github.com/golang/protobuf/proto latest\ngo: extracting github.com/armon/go-radix v1.0.0\ngo: downloading github.com/boltdb/bolt v1.3.1\ngo: finding github.com/Masterminds/vcs v1.13.1\ngo: downloading github.com/nightlyone/lockfile v0.0.0-20180618180623-0ad87eef1443\ngo: finding golang.org/x/sys/unix latest\ngo: downloading github.com/Masterminds/semver v1.4.2\ngo: downloading github.com/pkg/errors v0.8.1\ngo: downloading github.com/pelletier/go-toml v1.4.0\ngo: extracting github.com/Masterminds/semver v1.4.2\ngo: extracting github.com/nightlyone/lockfile v0.0.0-20180618180623-0ad87eef1443\ngo: extracting github.com/boltdb/bolt v1.3.1\ngo: extracting github.com/pkg/errors v0.8.1\ngo: extracting github.com/pelletier/go-toml v1.4.0\ngo: downloading github.com/Masterminds/vcs v1.13.1\ngo: finding golang.org/x/sync latest\ngo: extracting github.com/Masterminds/vcs v1.13.1\ngo: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58\ngo: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58\ngo: finding gopkg.in/yaml.v2 v2.2.2\ngo: finding golang.org/x/sys latest\ngo: downloading golang.org/x/sys v0.0.0-20190726091711-fc99dfbffb4e\ngo: finding github.com/golang/protobuf v1.3.2\ngo: downloading github.com/golang/protobuf v1.3.2\ngo: downloading gopkg.in/yaml.v2 v2.2.2\ngo: extracting github.com/golang/protobuf v1.3.2\ngo: extracting gopkg.in/yaml.v2 v2.2.2\ngo: extracting golang.org/x/sys v0.0.0-20190726091711-fc99dfbffb4e\ngo: finding github.com/davecgh/go-spew v1.1.1\ngo: finding github.com/BurntSushi/toml v0.3.1\ngo: finding gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405\n# github.com/golang/dep/gps\n../pki/go/pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:149:4: undefined: semver.Constraint", "stderr_lines": ["go: finding github.com/golang/dep/cmd/dep latest", "go: finding github.com/golang/dep/cmd latest", "go: finding github.com/golang/dep v0.5.4", "go: downloading github.com/golang/dep v0.5.4", "go: extracting github.com/golang/dep v0.5.4", "go: finding github.com/sdboyer/constext latest", "go: finding github.com/jmank88/nuts v0.3.0", "go: finding github.com/boltdb/bolt v1.3.1", "go: finding github.com/pkg/errors v0.8.1", "go: finding github.com/Masterminds/semver v1.4.2", "go: finding github.com/pelletier/go-toml v1.4.0", "go: finding github.com/armon/go-radix v1.0.0", "go: finding github.com/nightlyone/lockfile latest", "go: downloading github.com/sdboyer/constext v0.0.0-20170321163424-836a14457353", "go: extracting github.com/sdboyer/constext v0.0.0-20170321163424-836a14457353", "go: downloading github.com/jmank88/nuts v0.3.0", "go: extracting github.com/jmank88/nuts v0.3.0", "go: downloading github.com/armon/go-radix v1.0.0", "go: finding golang.org/x/sync/errgroup latest", "go: finding github.com/golang/protobuf/proto latest", "go: extracting github.com/armon/go-radix v1.0.0", "go: downloading github.com/boltdb/bolt v1.3.1", "go: finding github.com/Masterminds/vcs v1.13.1", "go: downloading github.com/nightlyone/lockfile v0.0.0-20180618180623-0ad87eef1443", "go: finding golang.org/x/sys/unix latest", "go: downloading github.com/Masterminds/semver v1.4.2", "go: downloading github.com/pkg/errors v0.8.1", "go: downloading github.com/pelletier/go-toml v1.4.0", "go: extracting github.com/Masterminds/semver v1.4.2", "go: extracting github.com/nightlyone/lockfile v0.0.0-20180618180623-0ad87eef1443", "go: extracting github.com/boltdb/bolt v1.3.1", "go: extracting github.com/pkg/errors v0.8.1", "go: extracting github.com/pelletier/go-toml v1.4.0", "go: downloading github.com/Masterminds/vcs v1.13.1", "go: finding golang.org/x/sync latest", "go: extracting github.com/Masterminds/vcs v1.13.1", "go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58", "go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58", "go: finding gopkg.in/yaml.v2 v2.2.2", "go: finding golang.org/x/sys latest", "go: downloading golang.org/x/sys v0.0.0-20190726091711-fc99dfbffb4e", "go: finding github.com/golang/protobuf v1.3.2", "go: downloading github.com/golang/protobuf v1.3.2", "go: downloading gopkg.in/yaml.v2 v2.2.2", "go: extracting github.com/golang/protobuf v1.3.2", "go: extracting gopkg.in/yaml.v2 v2.2.2", "go: extracting golang.org/x/sys v0.0.0-20190726091711-fc99dfbffb4e", "go: finding github.com/davecgh/go-spew v1.1.1", "go: finding github.com/BurntSushi/toml v0.3.1", "go: finding gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405", "# github.com/golang/dep/gps", "../pki/go/pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:149:4: undefined: semver.Constraint"], "stdout": "", "stdout_lines": []}
failed: [testNode] (item={u'url': u'golang.org/x/vgo', u'name': u'vgo'}) => {"ansible_loop_var": "item", "changed": false, "cmd": ["/usr/local/go/bin/go", "get", "-u", "golang.org/x/vgo"], "delta": "0:00:01.514037", "end": "2019-07-29 08:40:43.669417", "item": {"name": "vgo", "url": "golang.org/x/vgo"}, "msg": "non-zero return code", "rc": 2, "start": "2019-07-29 08:40:42.155380", "stderr": "go: finding golang.org/x/vgo latest\ngo: downloading golang.org/x/vgo v0.0.0-20180912184537-9d567625acf4\ngo: extracting golang.org/x/vgo v0.0.0-20180912184537-9d567625acf4\n# golang.org/x/vgo\n../pki/go/pkg/mod/golang.org/x/vgo@v0.0.0-20180912184537-9d567625acf4/main.go:31:2: undefined: main.Main", "stderr_lines": ["go: finding golang.org/x/vgo latest", "go: downloading golang.org/x/vgo v0.0.0-20180912184537-9d567625acf4", "go: extracting golang.org/x/vgo v0.0.0-20180912184537-9d567625acf4", "# golang.org/x/vgo", "../pki/go/pkg/mod/golang.org/x/vgo@v0.0.0-20180912184537-9d567625acf4/main.go:31:2: undefined: main.Main"], "stdout": "", "stdout_lines": []}```

The following indicates `vgo` should be removed from the defaults file.

I can have this adjusted for testing tonight for you.
```
golang.org/x/vgo\n../pki/go/pkg/mod/golang.org/x/vgo@v0.0.0-20180912184537-9d567625acf4/main.go:31:2: undefined: main.Main
```

Hi,
Thanks for the changes,
I have tested it again, but without success. Error is now:

`TASK [../roles/ansible-role-golang : Go-Lang | Run get commands] *******************************************************************************************************************************************
task path: ~/ansible/roles/ansible-role-golang/tasks/go-get.yml:21
failed: [testNode] (item={u'url': u'github.com/golang/dep/cmd/dep', u'name': u'dep'}) => {"ansible_loop_var": "item", "changed": false, "cmd": ["/usr/local/go/bin/go", "get", "-u", "github.com/golang/dep/cmd/dep"], "delta": "0:00:01.221634", "end": "2019-08-09 06:57:27.615809", "item": {"name": "dep", "url": "github.com/golang/dep/cmd/dep"}, "msg": "non-zero return code", "rc": 2, "start": "2019-08-09 06:57:26.394175", "stderr": "go: finding github.com/golang/dep/cmd/dep latest\ngo: finding github.com/golang/dep/cmd latest\ngo: finding github.com/sdboyer/constext latest\ngo: finding github.com/golang/protobuf/proto latest\ngo: finding github.com/nightlyone/lockfile latest\ngo: finding golang.org/x/sync/errgroup latest\ngo: finding golang.org/x/sync latest\ngo: finding golang.org/x/sys/unix latest\ngo: finding golang.org/x/sys latest\n# github.com/golang/dep/gps\n../pki/go/pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:149:4: undefined: semver.Constraint", "stderr_lines": ["go: finding github.com/golang/dep/cmd/dep latest", "go: finding github.com/golang/dep/cmd latest", "go: finding github.com/sdboyer/constext latest", "go: finding github.com/golang/protobuf/proto latest", "go: finding github.com/nightlyone/lockfile latest", "go: finding golang.org/x/sync/errgroup latest", "go: finding golang.org/x/sync latest", "go: finding golang.org/x/sys/unix latest", "go: finding golang.org/x/sys latest", "# github.com/golang/dep/gps", "../pki/go/pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:149:4: undefined: semver.Constraint"], "stdout": "", "stdout_lines": []}
`

--- a/defaults/main.yml
+++ b/defaults/main.yml
@@ -16,7 +16,5 @@ install_go_bootstrap: false
 go_build_script: make.bash
 
 # Go get
-go_get:
-- name: dep
-  url: github.com/golang/dep/cmd/dep
+go_get: []
 go_install: []

--- a/defaults/main.yml
+++ b/defaults/main.yml
@@ -19,6 +19,4 @@ go_build_script: make.bash
 go_get:
 - name: dep
   url: github.com/golang/dep/cmd/dep
-- name: vgo
-  url: golang.org/x/vgo
 go_install: []

--- a/tasks/setup.yml
+++ b/tasks/setup.yml
@@ -21,9 +21,9 @@
     fubarhouse_user_dir: "{{ shell_home_dir.stdout }}"
   when: fubarhouse_user_dir is not defined
 
-- name: "Go-Lang | Include OS-Specific tasks (CentOS)"
+- name: "Go-Lang | Include OS-Specific tasks (CentOS and Amazon)"
   include_tasks: tasks-CentOS.yml
-  when: ansible_distribution == "CentOS"
+  when: ansible_distribution == "CentOS" or ansible_distribution == "Amazon"
 
 - name: "Go-Lang | Include OS-Specific tasks (Darwin)"
   include_tasks: tasks-Darwin.yml
@@ -42,6 +42,7 @@
   when:
     - ansible_os_family == "RedHat"
     - ansible_distribution != "CentOS"
+    - ansible_distribution != "Amazon"
 
 - name: "Go-Lang | Define GO111MODULE"
   set_fact:

--- a/tests/test.yml
+++ b/tests/test.yml
@@ -11,6 +11,9 @@
     GOPATH: /home/travis/gopath
     GOROOT_BOOTSTRAP: /home/travis/goroot1.4
     GO111MODULE: "off"
+    go_get:
+      - name: dep
+        url: github.com/golang/dep/cmd/dep
 
   pre_tasks:
     - name: Check temporary directory permissions.

